File: exercise/21/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode *head = NULL;
        ListNode **p = &head;
        while (l1 != NULL && l2 != NULL) {
            if (l1->val < l2->val) {
                *p = l1;
                l1 = l1->next;
            } else {
                *p = l2;
                l2 = l2->next;
            }
            p = &((*p)->next);
        }
        if (l1 == NULL) {
            *p = l2;
        } else if (l2 == NULL) {
            *p = l1;
        }
        return head;
    }
};

File: exercise/21/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        fake = ListNode(0)
        now = fake
        while l1 and l2:
            if l1.val < l2.val:
                now.next = l1
                l1 = l1.next
            else:
                now.next = l2
                l2 = l2.next
            now = now.next
        if l1 is None:
            now.next = l2
        else:
            now.next = l1
        return fake.next
